lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java

```diff
@@ -75,7 +77,9 @@
     this.classpath = classpath;
   }
 
-  private void addYarnDriverConfiguration(final File driverFolder, final String jobId, final String jobSubmissionFolder)
+  private Configuration addYarnDriverConfiguration(final File driverFolder,
```

Comment:
The bridge needs to add the Java-Side before submitting. This involves creating & serializing the configuration for the Driver, as it depends on the chosen runtime.  `REEFImplementation` does not assume a pre-created Driver folder, for instance. The Bridge does, for efficiency reasons. Otherwise, we'd have to copy files around several times, which would slow down our tests (even more).

That said, there might be inefficiencies in how we do that, and code is most certainly duplicated between it an `REEFImplementation`.